### PR TITLE
[KYUUBI #2008] Support engine type and subdomain in kyuubi-ctl

### PR DESCRIPTION
**Pull request by cxzl25**
### _Why are the changes needed?_
Introduced engine type and subdomain, kyuubi-ctl does not support managing engine.
Add two engine-related parameters(--engine-type --engine-subdomain).

https://github.com/apache/incubator-kyuubi/issues/2008

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by cxzl25** [comment]
## Current
image

## Fix
image

**Review by yaooqinn — APPROVED** [review]
LGTM

File: kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/ServiceControlCliArgumentsSuite.scala [hunk 1]
```diff
@@ -67,6 +67,11 @@ class ServiceControlCliArgumentsSuite extends KyuubiFunSuite {
       }
       thread.start()
       thread.join()
+      // scalastyle:off println
+      println(logAppender.loggingEvents)
+      println("------")
+      println(searchString)
+      // scalastyle:on println
```
Comment (hddong) [inline kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/ServiceControlCliArgumentsSuite.scala:74]: Shall we remove those debug `println`?

Comment (cxzl25) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/ServiceControlCliArgumentsSuite.scala]: Sorry , I added it when I improved ut locally and forgot to remove it.

**Review by hddong — APPROVED** [review]

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**